### ISSUE TITLE
GPL-611 Upgrade SMRT link on test instance to v9.0

### ISSUE THREAD

**Issue by stevieing** [sanger/traction-service#463]
**User story**
As a developer we need to upgrade SMRT link on our test instance to v9.0 so we can test new version of sample sheets

**Who are the primary contacts for this story**
Steve / Harry

**Acceptance criteria**
To be considered successful the solution must allow:
- [ ] when I go to the SMRT link instance I should see v9.0 and be able to upload new versions of sample sheets
- [ ] ...

**Dependencies**
This story is blocked by the following dependencies:
- #<issue_no.>
- sanger/<repo>#<issue_no.>

**Additional context**

**Comment by KatyTaylor** [comment]
Update of where I've got to so far:

Made these changes to the ansible tasks, to deal specifically with upgrades: https://gitlab.internal.sanger.ac.uk/psd/psd-config/-/compare/master...GPL-nnn-upgrade-smrt-link

Log of attempts to upgrade:

1. In task "Upgrade SMRT Link by invoking the SMRT Link Installer", got error "Cannot write: No space left on device"
2. Removed smrtlink_8.0.0.80529.zip (1.6GB)
3. Re-ran the command while logged onto the host, with '--no-extract' flag added
4. Got error "TypeError: __init__() got an unexpected keyword argument 'encoding'"
    - Relevant logs in /opt/pacbio/smrtlink/userdata/log/smrtlink-analysisservices-gui/upgrade.log
    - The file with the failing line is /opt/pacbio/smrtlink/install/smrtlink-release_9.0.0.92188/bundles/smrtlink-analysisservices-gui/current/private/pacbio/smrtlink-analysisservices-gui/bin/dbctrl
    - From googling, “the encoding argument to Popen was only added in python3.6” (https://github.com/Maluuba/nlg-eval/issues/21)
    - on host, python3 --version gives 3.5.2
    - Tried to upgrade python to 3.6 using apt-get
    - Got 'Unable to locate package python3.6'
    - Installed python 3.6 from add-apt-repository ppa:deadsnakes/ppa
    - Installed fine, but python3 --version still gives the old 3.5 version, so 3.6 is not set as default

Looks like python 3.5 is automatically installed on Ubuntu 16.04

Possible next steps:
- Ask Carol if she had issues with python version, and ask what operating system the prod machine is on
- Upgrade our OS to Ubuntu 18.04 (the other version that SMRT Link supports) - think this comes with python 3.6 ready-installed
- Or, force the existing OS to use python 3.6 as default

**Comment by KatyTaylor** [comment]
Log from upgrading instance to Ubuntu 18.04 (from 16.04):

[SMRT_Link_upgrade_to_v9.txt](https://github.com/sanger/traction-service/files/5063183/SMRT_Link_upgrade_to_v9.txt)

**Comment by KatyTaylor** [comment]
Log from upgrading SMRT Link:

[smrt link upgrade log.txt](https://github.com/sanger/traction-service/files/5063741/smrt.link.upgrade.log.txt)

**Comment by KatyTaylor** [comment]
More details on what we did:

1. Ben asked Carol about the recent upgrade on prod, email chain pasted below:

We are in the process of upgrading out test SMRTLink instance from v8 to v9 for testing with Traction LIMS. We have a couple questions regarding the setup:
 
Did you have issues with python versions when upgrading SMRTLink and what version of python is production currently on?
> Nope. The main production host is farm5-pacbio01 where the default version of python is 2.7.17.... though other versions are available. 

> SMRT Link should come bundled with it's own python3 though e.g.

> [12:09:32 ces@farm5-pacbio01:/nfs/pacbio_data ] 
$ smrtlink/install/smrtlink-release_9.0.0.92188/bundles/smrttools/install/smrttools-release_9.0.0.92188/private/otherbins/thirdpartyall/bin/python3 --version
Python 3.7.3


What operating system is the production SMRTLink machine on?
> [12:09:43 ces@farm5-pacbio01:~ ] 
$ cat /etc/os-release 
NAME="Ubuntu"
VERSION="18.04.4 LTS (Bionic Beaver)"
ID=ubuntu
ID_LIKE=debian
PRETTY_NAME="Ubuntu 18.04.4 LTS"
VERSION_ID="18.04"
HOME_URL="https://www.ubuntu.com/"
SUPPORT_URL="https://help.ubuntu.com/"
BUG_REPORT_URL="https://bugs.launchpad.net/ubuntu/"
PRIVACY_POLICY_URL="https://www.ubuntu.com/legal/terms-and-policies/privacy-policy"
VERSION_CODENAME=bionic
UBUNTU_CODENAME=bionic

Does a test instance of SMRTLink currently exist or is SMRTLink upgraded directly on the production machine?
> I'm currently using seqfarm3-srv1 as the test machine - it's not a full setup e.g. no LDAP logins etc,  but allows for basic testing.

If there is a test instance of SMRTLink is it on the FCE?
> We still rely on LSF for job submissions so we don't use the raw FCE. We haven't run a test instance there for a little while now.


2. We upgraded our instance to Ubuntu over ssh, log attached in a previous comment

3. After this, upgrading the SMRT Link software worked fine, although I had to run it _without_ the `--no-extract` flag, having deleted the folder to which it was previously extracted. (If you try to run the command without that flag, and the folder exists, it will tell you which folder to delete).

4. I then ran the final task in ansible, which starts the server
